### ISSUE TITLE
TimeoutError when delivering with uproot-raw transformer

### ISSUE THREAD

**Issue by marctost11** [ssl-hep/ServiceX_frontend#385]
I am attempting to deliver a large dataset using the uproot-raw transformer. Smaller datasets have been working fine, but with larger ones, the delivery seems to fail.

Running on testing4, and on the website, all jobs show to be successfully transformed there. However, some of the files are not delivered to the local cache. Some files are delivered, and some only have .minio files delivered. After the message reporting that all transforms are done, the following error is displayed. 

I think something is going wrong with the moving of files to the local cache after being successfully transformed. Full traceback attached below:

---------------------------------------------------------------------------
TimeoutError                              Traceback (most recent call last)
/tmp/ipykernel_1360522/3399970052.py in <module>
     15         )
     16 
---> 17 data=deliver(sx_query)

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/servicex_client.py in deliver(config)
    105 
    106     elif config.General.Delivery == General.DeliveryEnum.LocalCache:
--> 107         results = group.as_files()
    108         return {obj.title: obj.file_list for obj in results}
    109 

~/.conda/envs/virtual_env/lib/python3.8/site-packages/make_it_sync/func_wrapper.py in wrapped_call(*args, **kwargs)
     61         @wraps(fn)
     62         def wrapped_call(*args, **kwargs):
---> 63             return _sync_version_of_function(fn, *args, **kwargs)
     64 
     65         return wrapped_call

~/.conda/envs/virtual_env/lib/python3.8/site-packages/make_it_sync/func_wrapper.py in _sync_version_of_function(fn, *args, **kwargs)
     24         future = exector.submit(get_data_wrapper, *args, **kwargs)
     25 
---> 26         return future.result()
     27 
     28 

~/.conda/envs/virtual_env/lib/python3.8/concurrent/futures/_base.py in result(self, timeout)
    442                     raise CancelledError()
    443                 elif self._state == FINISHED:
--> 444                     return self.__get_result()
    445                 else:
    446                     raise TimeoutError()

~/.conda/envs/virtual_env/lib/python3.8/concurrent/futures/_base.py in __get_result(self)
    387         if self._exception:
    388             try:
--> 389                 raise self._exception
    390             finally:
    391                 # Break a reference cycle with the exception in self._exception

~/.conda/envs/virtual_env/lib/python3.8/concurrent/futures/thread.py in run(self)
     55 
     56         try:
---> 57             result = self.fn(*self.args, **self.kwargs)
     58         except BaseException as exc:
     59             self.future.set_exception(exc)

~/.conda/envs/virtual_env/lib/python3.8/site-packages/make_it_sync/func_wrapper.py in get_data_wrapper(*args, **kwargs)
     19             asyncio.set_event_loop(loop)
     20             assert not loop.is_running()
---> 21             return loop.run_until_complete(fn(*args, **kwargs))
     22 
     23         exector = ThreadPoolExecutor(max_workers=1)

~/.conda/envs/virtual_env/lib/python3.8/asyncio/base_events.py in run_until_complete(self, future)
    614             raise RuntimeError('Event loop stopped before Future completed.')
    615 
--> 616         return future.result()
    617 
    618     def stop(self):

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/dataset_group.py in as_files_async(self, display_progress, provided_progress)
     87                 for d in self.datasets
     88             ]
---> 89             return await asyncio.gather(*self.tasks)
     90 
     91     as_files = make_sync(as_files_async)

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/query.py in as_files_async(self, display_progress, provided_progress)
    443         """
    444         with ExpandableProgress(display_progress, provided_progress) as progress:
--> 445             return await self.submit_and_download(signed_urls_only=False,
    446                                                   expandable_progress=progress)
    447 

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/query.py in submit_and_download(self, signed_urls_only, expandable_progress, dataset_group)
    264             signed_urls = []
    265             downloaded_files = []
--> 266             download_result = await download_files_task
    267             if signed_urls_only:
    268                 signed_urls = download_result

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/query.py in download_files(self, signed_urls_only, progress, download_progress, cached_record)
    431 
    432         # Now just wait until all of our tasks complete
--> 433         await asyncio.gather(*download_tasks)
    434         return result_uris
    435 

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/query.py in download_file(minio, filename, progress, download_progress, shorten_filename)
    373             shorten_filename: bool = False,
    374         ):
--> 375             downloaded_filename = await minio.download_file(
    376                 filename, self.download_path, shorten_filename=shorten_filename
    377             )

~/.conda/envs/virtual_env/lib/python3.8/site-packages/servicex/minio_adapter.py in download_file(self, object_name, local_dir, shorten_filename)
     94         )
     95 
---> 96         _ = await self.minio.fget_object(
     97             bucket_name=self.bucket, object_name=object_name, file_path=path.as_posix()
     98         )

~/.conda/envs/virtual_env/lib/python3.8/site-packages/miniopy_async/api.py in fget_object(self, bucket_name, object_name, file_path, request_headers, ssec, version_id, extra_query_params, tmp_file_path)
   1497                 )
   1498                 async with aiofile.async_open(tmp_file_path, "wb") as tmp_file:
-> 1499                     async for data in response.content.iter_chunked(n=1024 * 1024):
   1500                         await tmp_file.write(data)
   1501                 if os.path.exists(file_path):

~/.conda/envs/virtual_env/lib/python3.8/site-packages/aiohttp/streams.py in __anext__(self)
     37     async def __anext__(self) -> _T:
     38         try:
---> 39             rv = await self.read_func()
     40         except EofStream:
     41             raise StopAsyncIteration

~/.conda/envs/virtual_env/lib/python3.8/site-packages/aiohttp/streams.py in read(self, n)
    378         # without feeding any data
    379         while not self._buffer and not self._eof:
--> 380             await self._wait("read")
    381 
    382         return self._read_nowait(n)

~/.conda/envs/virtual_env/lib/python3.8/site-packages/aiohttp/streams.py in _wait(self, func_name)
    304             if self._timer:
    305                 with self._timer:
--> 306                     await waiter
    307             else:
    308                 await waiter

~/.conda/envs/virtual_env/lib/python3.8/site-packages/aiohttp/helpers.py in __exit__(self, exc_type, exc_val, exc_tb)
    654 
    655         if exc_type is asyncio.CancelledError and self._cancelled:
--> 656             raise asyncio.TimeoutError from None
    657         return None
    658 

TimeoutError:

**Comment by ponyisi** [comment]
Hi Marc, can you try this out again on testing1 (which has the latest uproot and codegen)?

**Comment by ponyisi** [comment]
Any problems recently @marctost11 ?

**Comment by marctost11** [comment]
No, biggest obstacle has been keeping an internet connect stable for extended periods of time, serviceX transformers are working well. MarcOn Oct 2, 2024, at 8:45 AM, ponyisi ***@***.***> wrote:﻿
Any problems recently @marctost11 ?
